### ISSUE TITLE
Heading-order may not correctly calculate level of heading element with aria-level

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#2971]
ARIA allows `<h1-6>` elements to have the `aria-level` attributes. This causes a potential bug in our [heading-order check](https://github.com/dequelabs/axe-core/blob/develop/lib/checks/navigation/heading-order-evaluate.js#L6-L18) as we only look at `aria-level` if the element has a `role=heading` attribute. Meaning that `<h1 aria-level="6">Goodbye</h1>` is calculated as a level 1 heading instead of a level 6 heading.

However, this doesn't work in all browsers. In modern browsers the `aria-level` is used to determine the heading level, but in IE 11 / JAWS the `aria-level` is ignored:

* FireFox / NVDA - "Heading level 6, Goodbye"
* Chrome / JAWS - "Heading level 6, Goodbye"
* IE 11 / JAWS - "Heading level 1, Goodbye"
* Safari / VO - "Heading level 6, Goodbye"
* Android Chrome / Voice Assistant - "Goodbye, Heading 6"

**Comment by straker** [comment]
Since everyone else is using the `aria-level` to calculate the level, we should update our code to do that.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base,
1.When we pass `aria-level="6" even it tagged in `h1` , and no `role` attribute is passing . It is announcing as heading 6.

![image](https://user-images.githubusercontent.com/47409579/128517080-06f5fc02-fa6f-41d3-bc19-e213236e6467.png)
2. we only look at aria-level if the element has a role=heading attribute. Meaning that `<h2 aria-level="5" and role=heading>hello</h2> ` is calculated as a level 5 heading instead of a level 2 heading.
![image](https://user-images.githubusercontent.com/47409579/128517372-508d1c81-769d-4b55-9c54-c57a6f99a82e.png)

**Comment by straker** [comment]
No docs
